mlflow/utils/autologging_utils.py

```diff
@@ -70,39 +36,27 @@ def log_fn_args_as_params(fn, args, kwargs, unlogged=[]):  # pylint: disable=W01
     :param unlogged: parameters not to be logged
     :return: None
     """
-    # all_default_values has length n, corresponding to values of the
-    # last n elements in all_param_names
-    pos_params, _, _, pos_defaults, kw_params, kw_defaults, _ = inspect.getfullargspec(fn)
-
-    kw_params = list(kw_params) if kw_params else []
-    pos_defaults = list(pos_defaults) if pos_defaults else []
-    all_param_names = pos_params + kw_params
-    all_default_values = pos_defaults + [kw_defaults[param] for param in kw_params]
-
-    # Checking if default values are present for logging. Known bug that getargspec will return an
-    # empty argspec for certain functions, despite the functions having an argspec.
-    if all_default_values is not None and len(all_default_values) > 0:
-        # Logging the default arguments not passed by the user
-        defaults = get_unspecified_default_args(args, kwargs, all_param_names, all_default_values)
-
-        for name in [name for name in defaults.keys() if name in unlogged]:
-            del defaults[name]
-        try_mlflow_log(mlflow.log_params, defaults)
-
-    # Logging the arguments passed by the user
-    args_dict = dict(
-        (param_name, param_val)
-        for param_name, param_val in zip(all_param_names, args)
-        if param_name not in unlogged
+    param_spec = inspect.signature(fn).parameters
```

Comment:
According to https://docs.python.org/3/library/inspect.html#inspect.getfullargspec, `inspect.signature()` is the recommended utility for method signature inspection in Python 3. `getfullargspec` is kept for compatibility with Python 2 code, which is no longer a concern for us. Further `inspect.getfullargspec` does not return accurate signature information for the latest versions of TensorFlow and Keras.

Accordingly, we drop in `inspect.signature()` in place of the previous `getfullargspec` and clean up the logic a bit.